include/rdma/fi_domain.h

```diff
@@ -223,6 +224,9 @@ struct fi_ops_domain {
 	int	(*query_atomic)(struct fid_domain *domain,
 			enum fi_datatype datatype, enum fi_op op,
 			struct fi_atomic_attr *attr, uint64_t flags);
+	int	(*query_collective)(struct fid_domain *domain,
+			struct fi_collective_attr *attr, enum fi_datatype datatype,
+			enum fi_op op, uint64_t flags);
```

Comment:
If we create a separate entry point, do these parameters make sense?  Or should we rework fi_collective_attr?

Reply:
yes we may want to slightly rework the `fi_collective_attr` and params here.  I think the `fi_collective_op` and the `fi_op` should swap, since the `fi_op` is only relevant for a subset of the collective operations. I would also consider moving the `fi_datatype` into the `fi_collective_attr` for the same reason.

Maybe something like this:
```
struct fi_collective_attr {
	enum fi_op 		op;
	enum fi_datatype 	datatype;
	struct fi_atomic_attr 	datatype_attr;
	size_t 			max_members;
	uint64_t 		mode;
};
```
```
int (*query_collective)(struct fid_domain *domain, enum fi_collective_op coll,
			struct fi_collective_attr *attr, uint64_t flags);
```

Reply:
I like those changes.

Note that I would keep the FI_VOID definition, so an app can still set the op to that rather than a specific value, even though it will be ignored.

If you can make these changes to the code, I will update the man page prior to the final release.

Reply:
so there is `FI_VOID` for the `fi_datatype`, but what is the equivalent for the `fi_op`?  barrier, for example, doesn't require a given op type.